### DISCUSSION TITLE
Sunbird-Ed / SunbirdEd-mobile-app & Sunbird ED Portal - Issue with Sunbird-PDF-Player

### DISCUSSION THREAD

**Discussion by chandu4002** [Sunbird-Knowlg/Community#140] [Bugs]
**Issue Description:**
**Unable to navigate to the last page of PDF using navigation button in Sunbird-PDF-Player.**
When we are accessing the PDF content in the Diksha Portal/mobile app, it will be open in sunbird-pdf-player so inside player we used to have navigation buttons to go the next/previous pages inside PDF. 
For suppose if we are having 4 pages in PDF, when we navigate from 3rd page to 4th page it is not navigating to the 4th page instead it is directly going to the end page.

But it is working fine, when we scroll the PDF normally without using navigation buttons.

**Note:** one thing we observed that whenever we zoom the PDF(using + button) in the Player, navigation works fine in both web and mobile.

Do id's: do_3131034749967728641957,do_31359465640938700812032

FYI, We are using the following npm library 
**"@project-sunbird/sunbird-pdf-player-v9": "4.8.1"**
In **Sunbird-Ed/SunbirdEd-mobile-app**, we are using the branch **release-5.1.0**

Here i am attaching the recordings for your reference.

https://github.com/Sunbird-Knowlg/Community/assets/129141695/3aae5007-c1a0-49dd-ad5f-8d2502c38b8a

https://github.com/Sunbird-Knowlg/Community/assets/129141695/85d2b78a-e433-466d-ae9c-8141ed8c88b6

**Comment by HarishGangula** [comment]
@chandu4002  Please use the latest version of the player and if the issue persists , let us know

**Reply by chandu4002** [reply]
@HarishGangula I will try installing latest version of library and let you know.
Thanks..

**Reply by chandu4002** [reply]
@HarishGangula
Tried installing the latest version of the player still issue not resolved.
**"@project-sunbird/sunbird-pdf-player-v9": "^5.5.0"**
Issue still exist.
Please find the attached recording for your reference.

https://github.com/Sunbird-Knowlg/Community/assets/129141695/1b9d5bf0-1baa-4fd8-b40a-d775f20e4abb

**Reply by HarishGangula** [reply]
@chandu4002  Is this specific to a pdf content or occurring for all the pdf content, let me know ?

**Reply by chandu4002** [reply]
@HarishGangula Issue is occurring for all the pdf content.

**Reply by HarishGangula** [reply]
@chandu4002 We are looking into it and get back to you

**Reply by HarishGangula** [reply]
@chandu4002  Can try with the latest version 5.6.0 and let us know

**Reply by chandu4002** [reply]
@HarishGangula Let me try and let you know.
Thanks..

**Reply by chandu4002** [reply]
@HarishGangula 
In **Sunbird-Ed/SunbirdEd-mobile-app**, we are using the branch **release-5.1.0**.
can we install the latest version of **player library(5.6.0)** in the above code base ?

**Reply by chandu4002** [reply]
@HarishGangula 
After installing latest **version (5.6.0)** of player in our code base, we are getting the following issue while running **npm run ionic-build**. 

Please find the attached screenshots

<img width="1512" alt="Screenshot 2023-06-27 at 3 19 17 PM" src="https://github.com/Sunbird-Knowlg/Community/assets/129141695/e4d03e67-a540-485c-b7b0-61d084fbd877">
<img width="1512" alt="Screenshot 2023-06-27 at 3 19 22 PM" src="https://github.com/Sunbird-Knowlg/Community/assets/129141695/1c04509a-58e2-403a-92f3-93c84645bb6d">



**Note:** your latest npm library Angular version is **14.2.5**. But in the project where we are installing this library the Angular version is **9.1.13**. Because of this version issue we are unable to build it or is it different issue?
Please let us know..

**Reply by HarishGangula** [reply]
@chandu4002  we have the web component version of the player also available please use it, if you have angular versioning issues

**Reply by chandu4002** [reply]
@HarishGangula 
we are using this library in ionic mobile application which is built on Angular version **9.1.13**. So we are unable to use this latest npm library in our project which is build on **Angular version 9** as you have updated this with Angular version **14.2.5**.

Can you do the fix for npm library **4.8.1** which is using **Angular version 9**.
Thanks.

**Reply by israr-trigyn** [reply]
@HarishGangula 

Please update if the fix can be done for **library 4.8.1**  as its using **angular version 9**, so that we can use the library in our projects which is also using **angular version 9**.

**Reply by HarishGangula** [reply]
You can cherry-pick the fix from the latest and apply it there or use the web component version of the player

**Reply by chandu4002** [reply]
@HarishGangula 
Okay thanks, we will check and let you know if anything is needed.

**Reply by HarishGangula** [reply]
@chandu4002  is it resolved, if yes? Please mark it as answered, otherwise, we can connect.
cc: @pallakartheekreddy @Krishnaj20

**Comment by chandu4002** [comment]
Issue got resolved.
